plugins/profiles.koplugin/main.lua

```diff
@@ -350,13 +360,6 @@ function Profiles:getProfileFromCurrentDocument(new_name)
         profile["embedded_fonts"] = self.ui.typeset.embedded_fonts
         profile["smooth_scaling"] = self.ui.typeset.smooth_scaling
     else
-        local trim_page_to_mode = { _("manual"), _("auto"), _("semi-auto"), _("none") }
-        local zoom_genus_to_mode = { _("manual"), _("rows"), _("columns"), _("content"), _("page") }
-        local zoom_type_to_mode = { _("height"), _("width"), _("full") }
```

Comment:
Did you try this without the translation?
Like:
```
local trim_page_to_mode = { "manual", "auto", "semi-auto", "none"}
```

Reply:
That's what I started from.
But it's a duplication of data that we usually try to avoid.

Reply:
For zooming you can use
```
ReaderZooming.zoom_mode_to_genus
ReaderZooming.zoom_mode_to_type
```

Reply:
Isn't it better to deduplicate ReaderZooming as well and have all that strings in one storage only - koptoptions?
Dispatcher would provide values/args/labels conversion when needed.
@NiLuJe ?

Reply:
Will this survive a change of UI language ? if you store (dunno if you do) such translated strings?

Reply:
In profile.lua I should not store translatable strings, it was a mistake.
In QM, Dispatcher shows translated "labels" correctly.

Reply:
> Isn't it better to deduplicate ReaderZooming as well and have all that strings in one storage only - koptoptions? Dispatcher would provide values/args/labels conversion when needed. @NiLuJe ?

I kinda lost the plot a bit, what are we talking about exactly?

Reply:
For creating a new profile with current pdf document settings we need arguments for zoom_mode settings.
Configurable stores values only.
I propose to ask koptoptions (from Dispatcher) to get arg from value.
Other way is to duplicate the value/arg table from koptoptions to profiles.lua.
You did so in https://github.com/koreader/koreader/pull/7463 (duplicated values/args tables).
I propose to think about the total deduplication and have one storage of the settings - koptoptions.
Dispatcher would provide values/args/labels conversion when needed.

Reply:
I don't necessarily know if it makes sense to make the strings in Dispatcher identical to those in the bottom panel or if a context should be used, but we haven't had any complaints yet afaik.

Reply:
That is to say, total deduplication could be sensible or it could be trying to be a bit too smart and I'm not quite sure which it is. :-)

Reply:
Unless I missed something (and I very well might, ReaderZooming is nightmare fuel, and I'm not familiar with ConfigDialog), but RZ itself is already using that table, so it's really duplication per-se. If the LUT is already used somewhere, no need to complicate matters more.

Reply:
In fact, RZ settings are so brittle that I'd vote *not* to rock the boat too much... On the other hand, that's also a very good reason to think about doing it better ;o).

TL;DR: If you feel like it, do what you feel is best.

Reply:
> if it makes sense to make the strings in Dispatcher identical to those in the bottom panel

They are not identical but just the same, both ConfigDialog and Dispatcher read them from coptoptions/koptoptions.

Reply:
> RZ itself is already using that table

Example of duplication:
https://github.com/koreader/koreader/blob/493a60faefbee6762b96600e6025ea15dc050c51/frontend/ui/data/koptoptions.lua#L285
https://github.com/koreader/koreader/blob/493a60faefbee6762b96600e6025ea15dc050c51/frontend/apps/reader/modules/readerzooming.lua#L40-L46
I proposed a new function `Dispatcher:getArgFromValue()`.

Reply:
The fact that page matches to 1 in the first case and 4 in the latter would make me want to keep this distinct to make that quirk very clear, but, again, not sure if relevant here ;).

(Also, I slept badly tonight, so I'm sorry if I'm slow and/or feel particularly unhelpful today ^^).

Reply:
> page matches to 1 in the first case and 4 in the latter

It's okay
https://github.com/koreader/koreader/blob/493a60faefbee6762b96600e6025ea15dc050c51/frontend/ui/data/koptoptions.lua#L281

Reply:
I don't really want to lose the RZ LUT (because RZ itself uses it, and the RZ settings are such a mess that having it like that in there helps readability immensely), regardless of what KoptOptions does, FWIW.

So, eeeh :/.

Reply:
I'm also not terribly fond of the idea of *Options or ConfigDialog being the source of truth, too, FWIW. It makes working with individual modules much murkier, i'd much rather each modules keep being the source of truth for their own stuff.

Reply:
Got it, thanks.

Reply:
Ok, i understand now :)

Please put a comment explaining what the function does, something like: "converts koptoptions values to args"